### PR TITLE
 PIP-72: Introduce Pulsar Interface Taxonomy: Audience and Stability Classification 

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

Introduce Pulsar Interface Taxonomy to provide guidance for developers and users to use Pulsar APIs.

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ReadOnlyCursor.java [hunk 1]
```diff
@@ -22,9 +22,13 @@
 
 import com.google.common.base.Predicate;
 import com.google.common.collect.Range;
+import org.apache.bookkeeper.common.annotation.InterfaceAudience;
```
Comment (eolivelli) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ReadOnlyCursor.java:25]: why are you using "org.apache.bookkeeper.common" and not "org.apache.pulsar.common" here and in other files  ?
it is a typo? or you did it for some reason ?

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ReadOnlyCursor.java (hunk 1 above)
Comment (sijie) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ReadOnlyCursor.java:25]: managed-ledger already used bookkeeper annotations. I am keeping it consistent.

Comment (eolivelli) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ReadOnlyCursor.java:25]: ok

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by hangc0276 — APPROVED** [review]

**Comment by hangc0276** [comment]
Good job.

**Pull request merged**